### PR TITLE
Fix build for Go 1.8

### PR DESCRIPTION
**Pull request by Serabe**
JS step is only run in Go 1.8

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -140,8 +142,10 @@ contributors:
 	git log --format='%aN <%aE>' | sort -fu > CONTRIBUTORS
 
 js:
+ifeq (go1.8, $(findstring go1.8, $(GO_VERSION)))
```
Comment (awalterschulze) [inline Makefile]: Why?

File: .travis.yml [hunk 2]
```diff
@@ -10,16 +10,18 @@ before_install:
...
   - 1.5.4
   - 1.6.3
   - 1.7.1
+  - 1.8
 
 matrix:
   allow_failures:
     - go: 1.5.4
     - go: 1.6.3
+    - go: 1.7.1
```
Comment (awalterschulze) [inline .travis.yml]: Does go1.7.1 actually fail?

File: Makefile (hunk 1 above)
Comment (Serabe) [inline Makefile]: Because gopherjs now requires 1.8

File: .travis.yml (hunk 2 above)
Comment (Serabe) [inline .travis.yml]: I am waiting for travis to actually check it.

**Comment by Serabe** [comment]
Updated. Hope Travis goes faster this time.

**Comment by Serabe** [comment]
For some reason, Go 1.8 changes the generated files :(

File: .travis.yml [hunk 3]
```diff
@@ -10,14 +10,15 @@ before_install:
 script:
   - PATH=/home/travis/bin:$PATH make buildserverall
   - echo $TRAVIS_GO_VERSION
-  - if [ "$TRAVIS_GO_VERSION" == 1.7.1 ] && [[ "$PROTOBUF_VERSION" == 3.1.0 ]]; then ! git status --porcelain | read || (git status; git diff; exit 1); fi
+  - if [ "$TRAVIS_GO_VERSION" == 1.8 ] && [[ "$PROTOBUF_VERSION" == 3.1.0 ]]; then ! git status --porcelain | read || (git status; git diff; exit 1); fi
 
 language: go
 
 go:
   - 1.5.4
```
Comment (awalterschulze) [inline .travis.yml]: I think we can remove 1.5.4
Also please update the Readme.md

File: Makefile (hunk 1 above)
Comment (awalterschulze) [inline Makefile]: Do you have a link to where it says this?

File: jsonpb/jsonpb_test_proto/more_test_objects.pb.go [hunk 4]
```diff
@@ -176,7 +176,7 @@ func init() { proto.RegisterFile("more_test_objects.proto", fileDescriptorMoreTe
 
 var fileDescriptorMoreTestObjects = []byte{
 	// 444 bytes of a gzipped FileDescriptorProto
-	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x09, 0x6e, 0x88, 0x02, 0xff, 0x8c, 0x94, 0xc1, 0x6b, 0xdb, 0x30,
+	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0x8c, 0x94, 0xc1, 0x6b, 0xdb, 0x30,
```
Comment (awalterschulze) [inline jsonpb/jsonpb_test_proto/more_test_objects.pb.go]: Very weird, but travis verifies that this is correct, so I guess its fine.

**Comment by Serabe** [comment]
Updated

**Comment by awalterschulze** [comment]
Thank you very much :)

**Comment by shurcooL** [comment]
> For some reason, Go 1.8 changes the generated files :(

@Serabe, that reason is https://golang.org/doc/go1.8#compress_gzip:

> The Writer now encodes a zero MTIME field when the Header.ModTime field is the zero value. In previous releases of Go, the Writer would encode a nonsensical value.

File: jsonpb/jsonpb_test_proto/more_test_objects.pb.go (hunk 4 above)
Comment (shurcooL) [inline jsonpb/jsonpb_test_proto/more_test_objects.pb.go]: See https://github.com/gogo/protobuf/pull/268#issuecomment-281537311.

File: Makefile (hunk 1 above)
Comment (shurcooL) [inline Makefile]: The latest version of GopherJS indeed requires Go 1.8. Here's a link that says it:

https://medium.com/gopherjs/gopherjs-1-8-1-is-released-b4473712ab95#.4espdfrxt

See second paragraph.

Comment (awalterschulze) [inline Makefile]: Thanks thats really good to know.
That also explains why there differences in the gzipped filedescriptor :)

**Pull request merged**